lib/file-source/src/file_watcher.rs

```diff
@@ -118,56 +107,21 @@ impl FileWatcher {
     pub fn read_line(&mut self, mut buffer: &mut Vec<u8>, max_size: usize) -> io::Result<usize> {
         //ensure buffer is re-initialized
         buffer.clear();
-        if let Some(ref mut reader) = self.reader {
```

Comment:
Has something changed that made it safe to assert this rather than `if let`?

Reply:
Yes, reader = None indicates a watcher is due to be deleted, and deletion almost immediately, before the next iteration of read_line.